toast-tk-rest-utils/src/main/java/io/toast/tk/core/rest/RestUtils.java

```diff
@@ -273,4 +304,58 @@ public static String downloadRepositoryAsWiki(String apiKey) {
 	public static void unRegisterAgent(String hostName) {
 		// NO-OP
 	}
+
+	private static PoolingHttpClientConnectionManager setHttpConnectionManager() throws NoSuchAlgorithmException, KeyStoreException, KeyManagementException {
+		// This connection manager must be used if more than one thread will
+	       // be using the HttpClient.
+			SSLContextBuilder builder = SSLContexts.custom();
+			builder.loadTrustMaterial(null, new TrustStrategy() {
+			    @Override
+			    public boolean isTrusted(X509Certificate[] chain, String authType)
+			            throws CertificateException {
+			        return true;
+			    }
+			});
+			
+			@SuppressWarnings("deprecation")
+			SSLConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(
+	                builder.build(),SSLConnectionSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER);
+			@SuppressWarnings("unused")
+			PlainConnectionSocketFactory pcsf = new PlainConnectionSocketFactory();
+			
+			Registry<ConnectionSocketFactory> socketFactoryRegistry = RegistryBuilder
+			        .<ConnectionSocketFactory> create().register("https", sslsf)
+			        .register("http", PlainConnectionSocketFactory.getSocketFactory())
+			        .build();
+
+	        PoolingHttpClientConnectionManager cm = new PoolingHttpClientConnectionManager(socketFactoryRegistry);
+	        cm.setMaxTotal(HTTP_CLIENT_MAX_POOL_SIZE);
+	        //cm.setDefaultSocketConfig( SocketConfig.custom().setSoKeepAlive( true ).setSoReuseAddress( true ).setSoTimeout( 3000 ).build() 
+	        //cm.setValidateAfterInactivity(1); // essai pour resoudre java.net.SocketException: Software caused connection abort: recv failed
+
+	        @SuppressWarnings("unused")
+			ConnectionKeepAliveStrategy myStrategy = new ConnectionKeepAliveStrategy() {
+
+	            public long getKeepAliveDuration(HttpResponse response, HttpContext context) {
+	                // Honor 'keep-alive' header
+	                HeaderElementIterator it = new BasicHeaderElementIterator(
+	                        response.headerIterator(HTTP.CONN_KEEP_ALIVE));
+	                while (it.hasNext()) {
+	                    HeaderElement he = it.nextElement();
+	                    String param = he.getName();
+	                    String value = he.getValue();
+	                    if (value != null && param.equalsIgnoreCase("timeout")) {
+	                        try {
+	                            return Long.parseLong(value) * 1000;
+	                        } catch(NumberFormatException ignore) {

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00108)